src/SIL.XForge.Identity/Controllers/Account/AccountController.cs

```diff
@@ -166,7 +179,92 @@ public async Task<IActionResult> Login(LoginInputModel model, string button)
             return View(vm);
         }
 
-        
+        /// <summary>
+        /// Show reset password page
+        /// </summary>
+        [HttpGet]
+        public async Task<IActionResult> ResetPassword(string token)
+        {
+            if (!string.IsNullOrEmpty(token))
+                return View();
+            else
+            {
+                LoginViewModel vm = await BuildLoginViewModelAsync("");
+                return View("Login", vm);
+            }
+        }
+
+        /// <summary>
+        /// Show forgot password page
+        /// </summary>
+        [HttpGet]
+        public async Task<IActionResult> ForgotPassword()
+        {
+            ForgotPasswordViewModel vm = BuildForgotPasswordViewModel();
+            return View("ForgotPassword", vm);
+        }
+
+        /// <summary>
+        /// Validate username or email and send email
+        /// </summary>
+        [HttpPost]
+        public async Task<IActionResult> ForgotPassword(ForgotPasswordViewModel model)
+        {
+            byte[] time = BitConverter.GetBytes(DateTime.UtcNow.ToBinary());
+            byte[] key = Guid.NewGuid().ToByteArray();
+            string resetPasswordKey = Convert.ToBase64String(time.Concat(key).ToArray());
+            UserEntity user = await _users.UpdateAsync(u => u.Username == model.UsernameOrEmail || u.Email == model.UsernameOrEmail, update => update
+            .Set(u => u.ResetPasswordKey, resetPasswordKey)
+            .Set(u => u.ResetPasswordExpirationDate, DateTime.Now.AddDays(7)));
+
+            if (user != null)
+            {
+                if (_environment.IsDevelopment())
```

Comment:
Instead of disabling this code in the development environment, I would create an `EmailService` class and a corresponding `IEmailService` interface in the SIL.XForge project. You could then mock the email service. You could use NSubstitute to verify that the correct data is passed to the EmailService.